tools/src/main/java/org/apache/kafka/trogdor/coordinator/Coordinator.java

```diff
@@ -130,16 +130,18 @@ public void run() {
                     List<Fault> toStart = new ArrayList<>();
                     lock.lock();
                     try {
+                        if (shutdown) {
```

Comment:
Maybe we could make `shutdown` volatile and move this check into the loop condition?

Reply:
Hmm.. I don't think having the check in the loop condition would work.  The issue is that you can have a sequence of events like this:

1. Thread1 calls `Coordinator#beginShutdown`.  We set `Coordinator#shutdown` to true, and call `Condition#signalAll`.  Since no threads are waiting on the condition, no threads are woken.
2. TrogdorCoordinatorThread calls `Condition#await` with a very long timeout.
3. Test times out because TrogdorCoordinatorThread never wakes up and observes that `shutdown` is true.

To prevent this sequence of events, we require beginShutdown to take the lock before setting `shutdown` to true.  And then TrogdorCoordinatorThread takes the lock, *and checks shutdown*, before calling await.

Reply:
Hmm.. I guess that makes sense. The pattern looks kind of funky, but I can't think of an obvious alternative. 

By the way, is it intentional that we don't reset `now` after the call to `await`? If so, it might be better to rename that variable to something like `startMs` or something like that to emphasize that it doesn't actually reflect the current time after calling `await`.

Reply:
I think it can be simplified a little bit by checking for shutdown only once, right after we take the lock and before we sleep or do anything else.

> By the way, is it intentional that we don't reset now after the call to await? If so, it might be better to rename that variable to startMs or something like that to emphasize that it doesn't actually reflect the current time after calling await.

No, this should be updated after the sleep.  Let me fix this.

Reply:
Thanks for the updates. Since we don't have the check after the `await` anymore, would it make sense to clear `pendingFaults` in `beginShutdown` so that we don't do any unneeded work after shutdown begins?

Reply:
ok